host-controller/src/main/java/org/jboss/as/host/controller/ManagedServerBootCmdFactory.java

```diff
@@ -239,6 +242,8 @@ public HostControllerEnvironment getHostControllerEnvironment() {
 
         command.add("-D[" + ManagedServer.getServerProcessName(serverName) + "]");
 
+        propagateJavaModuleParams(command);
```

Comment:
Please start a wildfly-dev list about how we are going to ultimately deal with this.

I don't mind putting in a breaking change like this as a workaround while we actually implement the final solution, but I don't think it's good to break things when we have no idea what the correct solution will be. That's just asking for a future crisis.

We want the common case where the servers are on the same jre as the host to be easy, but we also need to support servers running on a different jre.